lib/ansible/modules/cloud/amazon/ec2_instance.py

```diff
@@ -1084,6 +1086,16 @@ def discover_security_groups(group, groups, parent_vpc_id=None, subnet_id=None,
     return list(dict((g['GroupId'], g) for g in found_groups).values())
 
 
+def _supports_ebs_optimized(instance_type):
+    ebs_optimized_types = ['c1.xlarge', 'c3.xlarge', 'c3.2xlarge', 'c3.4xlarge',
+                           'g2.2xlarge', 'i2.xlarge', 'i2.2xlarge', 'i2.4xlarge',
+                           'm1.large', 'm1.xlarge', 'm2.2xlarge', 'm2.4xlarge',
+                           'm3.xlarge', 'm3.2xlarge', 'r3.xlarge', 'r3.2xlarge', 'r3.4xlarge']
```

Comment:
Perhaps I'm overlooking something, but this seems to not contain most newer generation instances? I believe there is some logic there, if you look at my Jinja code:
https://github.com/ansible/ansible/issues/55912

I'm only checking the instance class (first character) and generation (second character). For future proofing we can assume that newer instances in that same class also support this flag, for example if c5 supports EBS optimized, it's likely that c6 supports it too. 
https://ec2instances.info could possibly be used to generate a static list.

Reply:
See my earlier comment; newer instances don't need this, because the setting has no effect (EBS optimization is always on and cannot be disabled.) This setting is only useful on instances from the transitional period where EBS optimization was optional.

Reply:
Sorry, it's been some time since I was faced with this issue, thank you for clarifying.